mathics/core/expression.py

```diff
@@ -3317,3 +3353,106 @@ def construct(leaves):
 
 def string_list(head, leaves, evaluation):
     return atom_list_constructor(evaluation, head, "String")(leaves)
+
+
+def apply_N(expr, evaluation, prec=SymbolMachinePrecision):
+    """
+    This function applies recursively `N-value` rules for convert symbols, expressions and exact numbers
+    into Real numbers with certain precision.
+    Notice that `apply_N(expr, evaluation, prec)` is not equivalent in general to call
```

Comment:
This is related to my comment before. Maybe this _should_ be made to be equivalent so that the
caller doesn't have to worry. 

Imagine if we wanted to make a mathics "compiler" to bytecode. One of the bytecode operations would be "N".